torch/csrc/jit/passes/linear_to_matmul_add.cpp

```diff
@@ -0,0 +1,56 @@
+#include <algorithm>
+
+#include <torch/csrc/jit/ir/ir.h>
+#include <torch/csrc/jit/passes/linear_to_matmul_add.h>
+#include <torch/csrc/jit/passes/tensorexpr_fuser.h>
+
+namespace torch {
+namespace jit {
+
+namespace {
+
+void DecomposeLinearToMatmulAdd(Block *b) {
+  // TODO only do the expansion for GPU since CPU fusion isn't on by default
+  // (and thus may slow down instead)
+  std::vector<Node*> linear_nodes;
+  for (auto it = b->nodes().begin(); it != b->nodes().end(); it++) {
```

Comment:
nit: `(for  Node * n: b->nodes())` is the easier syntax when you are not modifying the graph in the body of the loop.